bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/devices/wifi/hub/TapoHubHandler.java

```diff
@@ -276,7 +283,7 @@ public List<TapoChildDeviceData> getChildDevices() {
 
     public TapoChildDeviceData getChild(String deviceSerial) {
         List<TapoChildDeviceData> childDeviceList = tapoChildsList.getChildDeviceList();
-        for (int i = 0; i <= childDeviceList.size(); i++) {
+        for (int i = 0; i < childDeviceList.size(); i++) {
```

Comment:
This seems to fix a NPE at the next line 288 or what made you change this?

Reply:
You're right, while testing I got an "Index out ouf bounds" error. This was the case when trying to add a RF-device without automatic discovery of the hub.

Reply:
Weird that this was not discovered before. But thanks for fixing it!